src/Microsoft.Build.Tasks.Git/GitDataReader/GitRepository.cs

```diff
@@ -209,6 +209,15 @@ public static GitRepository OpenRepository(GitRepositoryLocation location, GitEn
         /// <returns>Null if the HEAD tip reference can't be resolved.</returns>
         internal string? ReadSubmoduleHeadCommitSha(string submoduleWorkingDirectoryFullPath)
         {
+            // submodules don't usually have their own .git directories but can when Arcade uberclone
+            // was used.  Handle this case first since the other case throws.
```

Comment:
Ah, I was skeptical since I'd never seen it, but you're right that submodules may have a `.git` directory rather than `.git` file:

https://git-scm.com/docs/gitsubmodules#_forms

> "Old-form" submodule: A working directory with an embedded .git directory, and the tracking gitlink and .gitmodules entry in the superproject. This is typically found in repositories generated using older versions of Git.

https://git-scm.com/docs/git-submodule#Documentation/git-submodule.txt-absorbgitdirs

> A repository that was cloned independently and later added as a submodule or old setups have the submodules git directory inside the submodule instead of embedded into the superprojects git directory.

I don't think uberclone actually does make "real" submodules registered with Git--it just clones a repo within a repo--but I don't think this distinction matters. The reason we're in this code in the first place is that sourcelink is assuming a repo-within-a-repo is always a submodule, it seems.